src/gui/csvImport/CsvImportWidget.cpp

```diff
@@ -0,0 +1,296 @@
+    m_parserModel->setSkippedRows(rows);
+    updateTableview();
+}
+
+CsvImportWidget::~CsvImportWidget() {}
+
+void CsvImportWidget::configParser() {
+    m_parserModel->setBackslashSyntax(m_ui->checkBoxBackslash->isChecked());
+    m_parserModel->setComment(m_ui->comboBoxComment->currentText().at(0));
+    m_parserModel->setTextQualifier(m_ui->comboBoxTextQualifier->currentText().at(0));
+    m_parserModel->setCodec(m_ui->comboBoxCodec->currentText());
+    m_parserModel->setFieldSeparator(m_ui->comboBoxFieldSeparator->currentText().at(0));
+}
+
+void CsvImportWidget::updateTableview() {
+    m_ui->tableViewFields->resizeRowsToContents();
+    m_ui->tableViewFields->resizeColumnsToContents();
+
+    for (int c = 0; c < m_ui->tableViewFields->horizontalHeader()->count(); ++c)
+        m_ui->tableViewFields->horizontalHeader()->setSectionResizeMode(c, QHeaderView::Stretch);
+}
+
+void CsvImportWidget::updatePreview() {
+
+    m_ui->labelSizeRowsCols->setText(m_parserModel->getFileInfo());
+    m_ui->spinBoxSkip->setValue(0);
+    m_ui->spinBoxSkip->setMaximum(qMax(0, m_parserModel->rowCount() - 1));
+
+    int i;
+    QStringList list(tr("Not present in CSV file"));
+
+    for (i = 1; i < m_parserModel->getCsvCols(); i++) {
+        QString s = QString(tr("Column ")) + QString::number(i);
+        list << s;
+    }
+    m_comboModel->setStringList(list);
+
+    i=1;
+    Q_FOREACH (QComboBox* b, m_combos) {
+        if (i < m_parserModel->getCsvCols())
+            b->setCurrentIndex(i);
+        else
+            b->setCurrentIndex(0);
+        ++i;
+    }
+}
+
+void CsvImportWidget::load(const QString& filename, Database* const db) {
+    //QApplication::processEvents();
+    m_db = db;
+    m_parserModel->setFilename(filename);
+    m_ui->labelFilename->setText(filename);
+    Group* group = m_db->rootGroup();
+    group->setUuid(Uuid::random());
+    group->setNotes(tr("Imported from CSV file").append("\n").append(tr("Original data: ")) + filename);
+    parse();
+}
+
+void CsvImportWidget::parse() {
+    configParser();
+    QApplication::setOverrideCursor(Qt::WaitCursor);
+    //QApplication::processEvents();
+    bool good = m_parserModel->parse();
+    updatePreview();
+    QApplication::restoreOverrideCursor();
+    if (!good)
+        m_ui->messageWidget->showMessage(tr("Error(s) detected in CSV file !").append("\n")
+                                         .append(formatStatusText()), MessageWidget::Warning);
+    else
+        m_ui->messageWidget->setHidden(true);
+    QWidget::adjustSize();
+}
+
+
+QString CsvImportWidget::formatStatusText() const {
+    QString text = m_parserModel->getStatus();
+    int items = text.count('\n');
+    if (items > 2)
+        return text.section('\n', 0, 1)
+                .append("\n[").append(QString::number(items - 2))
+                .append(tr(" more messages skipped]"));
+    else
+        for (int i = 0; i < 2 - items; i++)
+            text.append(QString("\n"));
+        return text;
```

Comment:
I'm still getting the following warning:
```
/.../keepassxc/src/gui/csvImport/CsvImportWidget.cpp: In member function ‘QString CsvImportWidget::formatStatusText() const’:
/.../keepassxc/src/gui/csvImport/CsvImportWidget.cpp:191:5: warning: this ‘else’ clause does not guard... [-Wmisleading-indentation]
     else
     ^~~~
/.../keepassxc/src/gui/csvImport/CsvImportWidget.cpp:194:9: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘else’
         return text;
         ^~~~~~
```
It is generated by this code section. The only reason it works as intended is because it's the last statement in the function and the if block already contains a return. I would change it like this:

```
    if (items > 2) {
        return text.section('\n', 0, 1)
                   .append("\n[").append(QString::number(items - 2))
                   .append(tr(" more messages skipped]"));
    }

    for (int i = 0; i < 2 - items; i++) {
        text.append(QString("\n"));
    }

    return text;
```